lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -3091,17 +3093,20 @@ void request_stackTrace(const llvm::json::Object &request) {
     // This will always return an invalid thread when
     // libBacktraceRecording.dylib is not loaded or if there is no extended
     // backtrace.
-    lldb::SBThread queue_backtrace_thread =
-        thread.GetExtendedBacktraceThread("libdispatch");
+    lldb::SBThread queue_backtrace_thread;
+    if (g_dap.enable_display_extended_backtrace)
+      queue_backtrace_thread = thread.GetExtendedBacktraceThread("libdispatch");
```

Comment:
Maybe we should check SBProcess::GetNumExtendedBacktraceTypes() / GetExtendedBacktraceTypeAtIndex() instead of hard coding the `libdispatch`? I think that is configured by the target platform.

Would that solve the issue without needing an extra configuration flag?

Reply:
I still will want the option to disable this feature if it is coslty and not everyone needs it. But if all non mac platforms report "SBProcess::GetNumExtendedBacktraceTypes() == 0", then this can be a good work around. If we call this on mac, does `SBProcess::GetNumExtendedBacktraceTypes()` return 1 and `SBProcess::GetExtendedBacktraceTypeAtIndex(0)` return `"libdispatch"`? We definitely shouldn't be trying "libdispatch" on any non mac systems.

Reply:
Running an expression every time we stop is not a great thing to do and this is what is happening right now. Are we running one expression for every thread?

Reply:
According the Jeffrey's profiling, the penalty is up to 1 second in certain targets. I think that's a very reasonable wait time if that can improve the debugger experience, as long as it happens once per user action only in the cases where there's an extended backtrace via `GetNumExtendedBacktraceTypes`.
IIRC, not all stack traces are sent at once to the IDE. Only one stack trace gets sent and then, the user needs to expand individual threads to see more stack traces, in which case 1 second seems reasonable under the circumstances mentioned above.

Reply:
@ashgti, if you look at the profile trace, the hot path does not come from this call (`thread.GetExtendedBacktraceThread()`) but from `thread.GetCurrentExceptionBacktrace()` below.